code/datums/climate.dm

```diff
@@ -0,0 +1,209 @@
+/********************************************
+*                IN THIS FILE               *
+*          Climate Datum, Weather Datum     *
+*                                           *
+********************************************/
+
+#define PREDICTION_MINIMUM 4
+#define PREDICTION_MAXIMUM 10
+
+/datum/climate
+	var/datum/weather/current_weather
+	var/list/datum/weather/forecasts = list()
+	var/cycle_freq = list(3 MINUTES,6 MINUTES) //shortest possible time, longest possible time until next weather
+
+/datum/climate/New()
+	..()
+	if(current_weather)
+		forecast()
+	else
+		WARNING("Climate tried to forecast without a starting weather.")
+		message_admins("Climate tried to forecast without a starting weather.")
+
+/datum/climate/proc/forecast()
+	var/cycle = 1
+	forecasts = list(current_weather) //project based on current weather
+	while(forecasts.len <= PREDICTION_MINIMUM+1 && cycle <= PREDICTION_MAXIMUM)
+		var/datum/weather/W = forecasts[forecasts.len]
+		var/path = pickweight(W.next_weather)
+		if(path == W.type)
+			W.timeleft += round(rand(cycle_freq[1],cycle_freq[2]),SS_WAIT_WEATHER)
+		else
+			var/datum/weather/future = new path(src)
+			forecasts += future
+		cycle++
+	forecasts -= current_weather //remove it from our future weather
+
+/datum/climate/proc/tick()
+	if(!current_weather)
+		return
+	current_weather.tick()
+	if(current_weather.timeleft <= 0)
+		change_weather(forecasts[1])
+		forecasts -= forecasts[1]
+	if(forecasts.len < PREDICTION_MINIMUM)
+		forecast()
+
+/datum/climate/proc/change_weather(weather)
+	if(ispath(weather))
+		//We have been provided a path. Let's see if it's identical to the one we have.
+		if(ispath(weather, current_weather.type)) //This is a separate check so that we can have our warning work.
+			return //No need to change, this is our current type.
+		else
+			qdel(current_weather)
+			current_weather = new weather(src)
+			current_weather.execute()
+
+	else if(istype(weather,/datum/weather))
+		//We have been given a specific weather datum. It may be modified, so run it no matter what.
+		qdel(current_weather)
+		current_weather = weather
+		current_weather.execute()
+
+	else
+		WARNING("Change weather was called with [weather], neither a weather datum nor a path.")
+
+/datum/climate/snow
```

Comment:
Arctic may be a more fitting name here. \*shrug\*

Reply:
Also, it may be a good idea to assign a list of possible weathers to the climate.

Reply:
I'll take your name suggestion

Until there are new climates, I'm going to hold off on the other thing though. Currently, one weather has possibilities of others it leads into. Blacklisting may prove useful in a future where we do have alt climates though. We just don't have any reason for that to exist yet.